backend/src/main/java/com/htecgroup/skynest/service/impl/FolderServiceImpl.java

```diff
@@ -1,22 +1,41 @@
 package com.htecgroup.skynest.service.impl;
 
+import com.htecgroup.skynest.exception.folder.FolderAlreadyDeletedException;
 import com.htecgroup.skynest.exception.folder.FolderNotFoundException;
+import com.htecgroup.skynest.model.dto.FolderDto;
+import com.htecgroup.skynest.model.entity.ActionType;
 import com.htecgroup.skynest.model.entity.FolderEntity;
 import com.htecgroup.skynest.model.response.FolderResponse;
 import com.htecgroup.skynest.repository.FolderRepository;
+import com.htecgroup.skynest.service.ActionService;
 import com.htecgroup.skynest.service.FolderService;
 import lombok.AllArgsConstructor;
 import org.modelmapper.ModelMapper;
 import org.springframework.stereotype.Service;
 
+import java.util.Collections;
 import java.util.UUID;
 
 @Service
 @AllArgsConstructor
 public class FolderServiceImpl implements FolderService {
 
-  private ModelMapper modelMapper;
   private FolderRepository folderRepository;
+  private ModelMapper modelMapper;
+
+  private ActionService actionService;
+
+  @Override
+  public void removeFolder(UUID uuid) {
+    FolderDto folderDto = modelMapper.map(folderRepository.findFolderById(uuid), FolderDto.class);
+    if (folderDto.getDeletedOn() != null) {
```

Comment:
You can put 
`if (folderDto.getDeletedOn() != null)`
in a boolean method in FolderDto. It will give info if the folder is deleted or not, so the name can be isDeleted(). Maybe it can be reused in different place